server/src/main/java/org/elasticsearch/cluster/routing/allocation/ShardsAvailabilityHealthIndicatorService.java

```diff
@@ -83,27 +111,99 @@ public HealthIndicatorResult calculate(boolean includeDetails) {
+        return new HealthIndicatorResult(
+            name(),
+            component(),
+            status.getStatus(),
+            status.getSummary(),
+            status.getDetails(includeDetails),
+            status.getImpacts(),
+            status.getUserActions(includeDetails)
+        );
     }
 
-    private static class ShardAllocationCounts {
+    // TODO: #85572 Fill in help URLs once they are finalized
+    public static final UserAction.Definition ACTION_RESTORE_FROM_SNAPSHOT = new UserAction.Definition(
+        "restore_from_snapshot",
+        Explanations.Allocation.NO_COPIES,
+        null
+    );
+    public static final UserAction.Definition ACTION_CHECK_ALLOCATION_EXPLAIN_API = new UserAction.Definition(
+        "explain_allocations",
+        "Elasticsearch isn't allowed to allocate some shards from these indices to any of the nodes in the cluster. Diagnose the issue by "
+            + "calling the allocation explain api for an index [GET _cluster/allocation/explain]. Choose a node to which you expect a "
+            + "shard to be allocated, find this node in the node-by-node explanation, and address the reasons which prevent Elasticsearch "
+            + "from allocating a shard there.",
+        null
+    );
+    public static final UserAction.Definition ACTION_ENABLE_ALLOCATIONS = new UserAction.Definition(
+        "enable_allocations",
+        "Elasticsearch isn't allowed to allocate some shards from these indices because allocation for those shards has been disabled. "
+            + "Check that the ["
+            + EnableAllocationDecider.INDEX_ROUTING_ALLOCATION_ENABLE_SETTING.getKey()
+            + "] index settings and the ["
+            + EnableAllocationDecider.CLUSTER_ROUTING_ALLOCATION_ENABLE_SETTING.getKey()
+            + "] cluster setting are not set to ["
+            + EnableAllocationDecider.Allocation.NONE.toString().toLowerCase(Locale.getDefault())
+            + "].",
+        null
+    );
+    public static final UserAction.Definition ACTION_ENABLE_TIERS = new UserAction.Definition(
+        "enable_data_tiers",
+        "Elasticsearch isn't allowed to allocate shards from these indices because the indices expect to be allocated to data tier nodes, "
+            + "but none were found in the cluster. Add nodes with data tier roles ("
+            + DataTier.DATA_HOT
+            + ", "
+            + DataTier.DATA_WARM
+            + ", "
+            + DataTier.DATA_COLD
+            + ", etc.) to the cluster.",
+        null
+    );
+    public static final UserAction.Definition ACTION_SHARD_LIMIT = new UserAction.Definition(
+        "increase_shard_limit",
+        "Elasticsearch isn't allowed to allocate some shards from these indices to any of the nodes in its data tier because each node in "
+            + "the tier has reached its shard limit. Increase the values for the ["
```

Comment:
Can we include what the shard limit is in the message?

Reply:
This is a bit of a weird one because of how the messages are currently laid out. The message displayed to a user is part of an immutable action definition. If we include an index scope setting value in the message for the user action we would need to create a new user action entry for each index affected by this issue. This would be the only way to have each message carry the value for each index.

Furthermore, if the number of indices that this problem affects is large, the response would be substantially larger if each affected index gets its own user action block.

Reply:
I think it'll usually be the case of one tier needing a scale-up. Could we specify which tier(s) need it? The current message is too generic to be actionable IMO

Reply:
Also, could you please list a complete response in the PR description? (that'd be a life saver when processing all the diagnosis we do here :) - and we could lift and shift that into the docs)

Reply:
Tier scaling change in [6042c00](https://github.com/elastic/elasticsearch/pull/85071/commits/6042c00148846e3c40fd2976567ee283489460b8) and an example of a response was added to the PR description - I'll work on getting the remaining user actions added in as well